pkgs/by-name/ma/marcel/package.nix

```diff
@@ -0,0 +1,45 @@
+{ lib
+, fetchFromGitHub
+, python3Packages
+, bash
+}:
+
+python3Packages.buildPythonApplication rec {
+  pname = "marcel";
+  version = "0.22.2";
+  pyproject = true;
+
+  src = fetchFromGitHub {
+    owner = "geophile";
+    repo = "marcel";
+    rev = "refs/tags/v${version}";
+    hash = "sha256-CiAIY4qXv5V2VOsi+vat7OvFcmeFpIrmHCfqlY+JRXc=";
+  };
+
+  propagatedBuildInputs = [
+    bash
+  ] ++ (with python3Packages; [
+    dill
+    psutil
+    setuptools
```

Comment:
setuptools should be in `nativeBuildInputs` not `propagatedBuildInputs` as it is just used to build the package but not required afterwards. that should be fixed, the rest of the comment is just nits.


bash can just be in `buildInputs` and can use pythonPath for the rest rather than propagatedBuildInputs. propagatedBuildInputs would be needed if you were importing marcel in some other python package, but since this is an application, you're not, so `pythonPath` is sufficient.

this whole thing could be written as:
```nix
  nativeBuildInputs = with python3Packages; [
    setuptools
  ];

  buildInputs = [
    bash
  ];

  pythonPath = with python3Packages; [
    dill
    psutil
  ];
```